fud2/tests/tests.rs

```diff
@@ -70,23 +84,77 @@ fn req_slug(driver: &Driver, req: &Request) -> String {
     desc
 }
 
-fn test_emit(driver: &Driver, req: Request) {
+fn test_emit(driver: &Driver, req: Request, tag: &str) {
     let desc = req_desc(driver, &req);
     let slug = req_slug(driver, &req);
     let ninja = emit_ninja(driver, req);
     insta::with_settings!({
         description => desc,
         omit_expression => true,
-        snapshot_suffix => slug,
+        snapshot_suffix => format!("{slug}{tag}"),
     }, {
         insta::assert_snapshot!(ninja);
     });
 }
 
+#[test]
+fn all_ops() {
+    let driver = test_driver();
+    for op in driver.ops.values() {
+        let req = fud_core::exec::Request {
+            start_file: None,
+            start_state: op.input,
+            end_file: None,
+            end_state: op.output,
+            through: vec![],
+            workdir: ".".into(),
+        };
+        test_emit(&driver, req, &format!("__op_{}", op.name));
+    }
+}
```

Comment:
No real comment here; just highlighting the critical part of this testing strategy (i.e., loop over all the ops and construct a test for just that op alone, using its input & output state) for other readers.